// baseui/themes/dark-theme/color-semantic-tokens.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _colorFoundationTokens = _interopRequireDefault(require("./color-foundation-tokens"));
var _util = require("../../styles/util");
var _colorPrimitiveTokens = require("../../tokens/color-primitive-tokens");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
var _default = (foundation = _colorFoundationTokens.default) => {
  const core = {
    // Background
    backgroundPrimary: foundation.primaryB,
    backgroundSecondary: _colorPrimitiveTokens.primitiveDarkColors.gray100Dark,
    backgroundTertiary: _colorPrimitiveTokens.primitiveDarkColors.gray200Dark,
    backgroundInversePrimary: _colorPrimitiveTokens.primitiveDarkColors.gray800Dark,
    backgroundInverseSecondary: _colorPrimitiveTokens.primitiveDarkColors.gray700Dark,
    // Content
    contentPrimary: foundation.primaryA,
    contentSecondary: _colorPrimitiveTokens.primitiveDarkColors.gray800Dark,
    contentTertiary: _colorPrimitiveTokens.primitiveDarkColors.gray700Dark,
    contentInversePrimary: _colorPrimitiveTokens.primitiveDarkColors.black,
    contentInverseSecondary: _colorPrimitiveTokens.primitiveDarkColors.gray200Dark,
    contentInverseTertiary: _colorPrimitiveTokens.primitiveDarkColors.gray300Dark,
    // Border
    borderOpaque: _colorPrimitiveTokens.primitiveDarkColors.gray100Dark,
    borderTransparent: (0, _util.hexToRgb)(foundation.primaryA, '0.08') || '',
    borderSelected: foundation.primaryA,
    borderInverseOpaque: _colorPrimitiveTokens.primitiveDarkColors.gray300Dark,
    borderInverseTransparent: (0, _util.hexToRgb)(foundation.primaryB, '0.2') || '',
    borderInverseSelected: foundation.primaryB
  };
  const coreExtensions = {
    // Backgrounds
    backgroundStateDisabled: _colorPrimitiveTokens.primitiveDarkColors.gray100Dark,
    backgroundOverlay: (0, _util.hexToRgb)(_colorPrimitiveTokens.primitiveDarkColors.black, '0.7') || '',
    backgroundOverlayArt: (0, _util.hexToRgb)(_colorPrimitiveTokens.primitiveDarkColors.black, '0.16') || '',
    backgroundAccent: foundation.accent,
    backgroundNegative: foundation.negative,
    backgroundWarning: foundation.warning,
    backgroundPositive: foundation.positive,
    backgroundAccentLight: _colorPrimitiveTokens.primitiveDarkColors.blue100Dark,
    backgroundPositiveLight: _colorPrimitiveTokens.primitiveDarkColors.green100Dark,
    backgroundNegativeLight: _colorPrimitiveTokens.primitiveDarkColors.red100Dark,
    backgroundWarningLight: _colorPrimitiveTokens.primitiveDarkColors.yellow100Dark,
    backgroundAlwaysDark: _colorPrimitiveTokens.primitiveDarkColors.gray100Dark,
    backgroundAlwaysLight: _colorPrimitiveTokens.primitiveDarkColors.gray900Dark,
    // Content
    contentStateDisabled: _colorPrimitiveTokens.primitiveDarkColors.gray400Dark,
    contentAccent: _colorPrimitiveTokens.primitiveDarkColors.blue600Dark,
    contentOnColor: _colorPrimitiveTokens.primitiveDarkColors.gray900Dark,
    contentOnColorInverse: _colorPrimitiveTokens.primitiveDarkColors.black,
    contentNegative: _colorPrimitiveTokens.primitiveDarkColors.red600Dark,
    contentWarning: _colorPrimitiveTokens.primitiveDarkColors.yellow600Dark,
    contentPositive: _colorPrimitiveTokens.primitiveDarkColors.green600Dark,
    // Border
    borderStateDisabled: _colorPrimitiveTokens.primitiveDarkColors.gray100Dark,
    borderAccent: _colorPrimitiveTokens.primitiveDarkColors.blue500Dark,
    borderAccentLight: _colorPrimitiveTokens.primitiveDarkColors.blue400Dark,
    borderNegative: _colorPrimitiveTokens.primitiveDarkColors.red500Dark,
    borderNegativeLight: _colorPrimitiveTokens.primitiveDarkColors.red400Dark,
    borderWarning: _colorPrimitiveTokens.primitiveDarkColors.yellow500Dark,
    borderWarningLight: _colorPrimitiveTokens.primitiveDarkColors.yellow400Dark,
    borderPositive: _colorPrimitiveTokens.primitiveDarkColors.green500Dark,
    borderPositiveLight: _colorPrimitiveTokens.primitiveDarkColors.green400Dark,
    // Programs
    safety: _colorPrimitiveTokens.primitiveLightColors.blue600,
    eatsGreen400: _colorPrimitiveTokens.primitiveLightColors.green600,
    freightBlue400: _colorPrimitiveTokens.primitiveLightColors.cobalt400,
    rewardsTier1: _colorPrimitiveTokens.primitiveLightColors.blue600,
    rewardsTier2: _colorPrimitiveTokens.primitiveLightColors.yellow300,
    rewardsTier3: _colorPrimitiveTokens.primitiveLightColors.platinum400,
    rewardsTier4: _colorPrimitiveTokens.primitiveLightColors.black,
    membership: _colorPrimitiveTokens.primitiveLightColors.yellow600
  };
  const deprecated = {
    jumpRed400: _colorPrimitiveTokens.primitiveLightColors.red600,
    backgroundOverlayLight: coreExtensions.backgroundOverlay,
    backgroundOverlayDark: coreExtensions.backgroundOverlay,
    backgroundLightAccent: coreExtensions.backgroundAccentLight,
    backgroundLightPositive: coreExtensions.backgroundPositiveLight,
    backgroundLightWarning: coreExtensions.backgroundWarningLight,
    backgroundLightNegative: coreExtensions.backgroundNegativeLight
  };
  return {
    ...core,
    ...coreExtensions,
    ...deprecated
  };
};
exports.default = _default;
