// baseui/file-uploader/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FILE_STATUS_TO_COLOR_MAP = exports.FILE_STATUS = exports.ARIA_LIVE_TIMEOUT_MS = exports.ARIA_LIVE_ELEMENT_ID = void 0;
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

const FILE_STATUS = exports.FILE_STATUS = {
  added: 'added',
  error: 'error',
  processed: 'processed'
};
const FILE_STATUS_TO_COLOR_MAP = $theme => ({
  [FILE_STATUS.added]: $theme.colors.accent,
  [FILE_STATUS.error]: $theme.colors.negative,
  [FILE_STATUS.processed]: $theme.colors.positive
});
exports.FILE_STATUS_TO_COLOR_MAP = FILE_STATUS_TO_COLOR_MAP;
let ARIA_LIVE_ELEMENT_ID = exports.ARIA_LIVE_ELEMENT_ID = /*#__PURE__*/function (ARIA_LIVE_ELEMENT_ID) {
  ARIA_LIVE_ELEMENT_ID["ADDITION"] = "file-uploader-addition";
  ARIA_LIVE_ELEMENT_ID["REMOVAL"] = "file-uploader-removal";
  return ARIA_LIVE_ELEMENT_ID;
}({});
const ARIA_LIVE_TIMEOUT_MS = exports.ARIA_LIVE_TIMEOUT_MS = 5000;
