// baseui/themes/light-theme/color-semantic-tokens.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _colorFoundationTokens = _interopRequireDefault(require("./color-foundation-tokens"));
var _util = require("../../styles/util");
var _colorPrimitiveTokens = _interopRequireDefault(require("../../tokens/color-primitive-tokens"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
var _default = (
// themePrimitives or foundation colors
foundation = _colorFoundationTokens.default) => {
  const core = {
    // Background
    backgroundPrimary: foundation.primaryB,
    backgroundSecondary: _colorPrimitiveTokens.default.gray50,
    backgroundTertiary: _colorPrimitiveTokens.default.gray100,
    backgroundInversePrimary: foundation.primaryA,
    backgroundInverseSecondary: _colorPrimitiveTokens.default.gray900,
    // Content
    contentPrimary: foundation.primaryA,
    contentSecondary: _colorPrimitiveTokens.default.gray800,
    contentTertiary: _colorPrimitiveTokens.default.gray700,
    contentInversePrimary: foundation.primaryB,
    contentInverseSecondary: _colorPrimitiveTokens.default.gray200,
    contentInverseTertiary: _colorPrimitiveTokens.default.gray400,
    // Border
    borderOpaque: _colorPrimitiveTokens.default.gray50,
    borderTransparent: (0, _util.hexToRgb)(foundation.primaryA, '0.08') || (0, _util.hexToRgb)(_colorFoundationTokens.default.primaryA, '0.08') || '',
    borderSelected: foundation.primaryA,
    borderInverseOpaque: _colorPrimitiveTokens.default.gray800,
    borderInverseTransparent: (0, _util.hexToRgb)(foundation.primaryB, '0.2') || (0, _util.hexToRgb)(_colorFoundationTokens.default.primaryB, '0.2') || '',
    borderInverseSelected: foundation.primaryB
  };
  const coreExtensions = {
    // Backgrounds
    backgroundStateDisabled: _colorPrimitiveTokens.default.gray50,
    backgroundOverlay: (0, _util.hexToRgb)(_colorPrimitiveTokens.default.black, '0.5') || '',
    backgroundOverlayArt: (0, _util.hexToRgb)(_colorPrimitiveTokens.default.black, '0.00') || '',
    backgroundAccent: foundation.accent,
    backgroundNegative: foundation.negative,
    backgroundWarning: foundation.warning,
    backgroundPositive: foundation.positive,
    backgroundAccentLight: _colorPrimitiveTokens.default.blue50,
    backgroundNegativeLight: _colorPrimitiveTokens.default.red50,
    backgroundWarningLight: _colorPrimitiveTokens.default.yellow50,
    backgroundPositiveLight: _colorPrimitiveTokens.default.green50,
    backgroundAlwaysDark: _colorPrimitiveTokens.default.black,
    backgroundAlwaysLight: _colorPrimitiveTokens.default.white,
    // Content
    contentStateDisabled: _colorPrimitiveTokens.default.gray400,
    contentOnColor: _colorPrimitiveTokens.default.white,
    contentOnColorInverse: _colorPrimitiveTokens.default.black,
    contentAccent: _colorPrimitiveTokens.default.blue600,
    contentNegative: _colorPrimitiveTokens.default.red600,
    contentWarning: _colorPrimitiveTokens.default.yellow600,
    contentPositive: _colorPrimitiveTokens.default.green600,
    // Border
    borderStateDisabled: _colorPrimitiveTokens.default.gray50,
    borderAccent: _colorPrimitiveTokens.default.blue600,
    borderAccentLight: _colorPrimitiveTokens.default.blue300,
    borderNegative: _colorPrimitiveTokens.default.red600,
    borderNegativeLight: _colorPrimitiveTokens.default.red300,
    borderWarning: _colorPrimitiveTokens.default.yellow600,
    borderWarningLight: _colorPrimitiveTokens.default.yellow200,
    borderPositive: _colorPrimitiveTokens.default.green600,
    borderPositiveLight: _colorPrimitiveTokens.default.green300,
    // Programs
    safety: _colorPrimitiveTokens.default.blue600,
    eatsGreen400: _colorPrimitiveTokens.default.green600,
    freightBlue400: _colorPrimitiveTokens.default.cobalt400,
    rewardsTier1: _colorPrimitiveTokens.default.blue600,
    rewardsTier2: _colorPrimitiveTokens.default.yellow300,
    rewardsTier3: _colorPrimitiveTokens.default.platinum400,
    rewardsTier4: _colorPrimitiveTokens.default.black,
    membership: _colorPrimitiveTokens.default.yellow600
  };
  /** @deprecated these tokens are deprecated */
  const deprecated = {
    jumpRed400: _colorPrimitiveTokens.default.red400,
    backgroundOverlayLight: coreExtensions.backgroundOverlay,
    backgroundOverlayDark: coreExtensions.backgroundOverlay,
    backgroundLightAccent: coreExtensions.backgroundAccentLight,
    backgroundLightPositive: coreExtensions.backgroundPositiveLight,
    backgroundLightWarning: coreExtensions.backgroundWarningLight,
    backgroundLightNegative: coreExtensions.backgroundNegativeLight
  };
  return {
    ...core,
    ...coreExtensions,
    ...deprecated
  };
};
exports.default = _default;
